sys/include/sema.h

```diff
@@ -129,22 +117,28 @@ int sema_wait_timed(sema_t *sema, uint64_t timeout);
  * @param[in]  sema A semaphore.
  *
  * @return  0 on success
- * @return  -EINVAL, if semaphore is invalid.
- * @return  -ECANCELED, if the semaphore was destroyed.
+ * @return  -1 on error (i.e. semaphore was destroyed)
```

Comment:
Here and in other places: Changing these return values changes the behavior of the POSIX wrapper (it uses the result to set `errno`, as specified by the POSIX specifications). Either we drop this behavior (and don't care about POSIX compatibility and also fix the POSIX wrapper accordingly) or these values needs to stay the same.